src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2021,13 +2024,24 @@ WorkerExtendedOpNode(MultiExtendedOp *originalOpNode,
 
 	if (enableLimitPushdown)
 	{
+		List *newTargetEntryListForSortClauses = NIL;
+
 		/* if we can push down the limit, also set related fields */
 		workerExtendedOpNode->limitCount = WorkerLimitCount(originalOpNode,
 															groupedByDisjointPartitionColumn);
 		workerExtendedOpNode->sortClauseList =
 			WorkerSortClauseList(originalOpNode, groupedByDisjointPartitionColumn);
+
+		newTargetEntryListForSortClauses = GenerateNewTargetEntriesForSortClauses(
```

Comment:
targetProjectionNumber is used to determine value of ```resno``` field.  Having an out variable can be worked out as you suggested but in any case we need to pass in something so that we can use correct resno value.